app/services/remote_identities/create_service.rb

```diff
@@ -32,35 +32,31 @@ module RemoteIdentities
   class CreateService
     attr_reader :user, :model
 
-    def self.call(user:, oauth_config:, oauth_client_token:)
-      new(user:, oauth_config:, oauth_client_token:).call
+    def self.call(user:, integration:, token:)
+      new(user:, integration:, token:).call
     end
 
-    def initialize(user:, oauth_config:, oauth_client_token:)
+    def initialize(user:, integration:, token:)
       @user = user
-      @oauth_config = oauth_config
-      @oauth_client_token = oauth_client_token
+      @integration = integration
+      @token = token
 
-      @model = RemoteIdentity.find_or_initialize_by(user:, oauth_client: oauth_config.oauth_client)
+      @model = RemoteIdentity.find_or_initialize_by(user:, auth_source: token.auth_source, integration:)
       @result = ServiceResult.success(result: @model, errors: @model.errors)
     end
 
     def call
-      @model.origin_user_id = @oauth_config.extract_origin_user_id(@oauth_client_token)
+      @model.origin_user_id = @integration.extract_origin_user_id(@token)
       if @model.save
-        emit_event(@oauth_config.oauth_client.integration)
+        OpenProject::Notifications.send(
+          OpenProject::Events::REMOTE_IDENTITY_CREATED,
```

Comment:
:yellow_circle:  I am wondering whether this code won't be run every time that we fetch a token for the storage.

I see nothing limiting us to the `create` case here, so also all subsequent updates (possibly with no changes) will trigger the `REMOTE_IDENTITY_CREATED` event.

Am I missing something? Is this intended?

Reply:
Exactly. I think under normal circumstances the event should be fired once upon creation.